### ISSUE TITLE
cmd/go: repeating packages for "go test" is racy

### ISSUE THREAD

**Issue by tv42** [golang/go#13005]
I accidentally typoed `go test . ./...` in a build script and noticed an interesting trap: the first `.` may be executing tests while the second `.` from `./...` tries to open the same executable for writing, leading into a "text file is busy" message, if the timing is just right.

I would have expected either different temp file names, or `go test` to only run test for a package once, even if it was listed twice.

It was pointed out to me that `go test fmt fmt fmt` already runs the `fmt` tests just once, so this probably is about the `...` logic not participating in the uniquify.

**Comment by tv42** [comment]
```console
[0 tv@brute ~/go/src/bazil.org/bazil/kv]$ go test . ./...
fork/exec /home/tv/tmp/go-build973088303/bazil.org/bazil/kv/_test/kv.test: text file busy
FAIL	bazil.org/bazil/kv	0.001s
fork/exec /home/tv/tmp/go-build973088303/bazil.org/bazil/kv/_test/kv.test: no such file or directory
FAIL	bazil.org/bazil/kv	0.001s
ok  	bazil.org/bazil/kv/kvfiles	0.002s
?   	bazil.org/bazil/kv/kvmock	[no test files]
ok  	bazil.org/bazil/kv/kvmulti	0.002s
?   	bazil.org/bazil/kv/kvpeer	[no test files]
ok  	bazil.org/bazil/kv/untrusted	0.004s
```

**Comment by adg** [comment]
Interesting find. I think the `go` command should normalize the list of input packages before doing anything.

**Comment by nodirt** [comment]
I'll take this

**Comment by nodirt** [comment]
Fixed by [CL 16136](https://go-review.googlesource.com/#/c/16136/)

**Comment by adg** [comment]
Thanks @nodirt! Sorry I forgot to ask you to add the "Fixes #13005" line to the change description.

**Comment by nodirt** [comment]
I forgot to add it :)
